bson/primitive/decimal_test.go

```diff
@@ -160,15 +160,34 @@ func TestDecimal128_JSON(t *testing.T) {
 		assert.Equal(t, decimal.h, got.h, "expected h: %v got: %v", decimal.h, got.h)
 		assert.Equal(t, decimal.l, got.l, "expected l: %v got: %v", decimal.l, got.l)
 	})
-	t.Run("Unmarshal", func(t *testing.T) {
+	t.Run("unmarshal extendedJSON", func(t *testing.T) {
+		want := NewDecimal128(0x3040000000000000, 12345)
+		extJSON := fmt.Sprintf(`{"$numberDecimal": %q}`, want.String())
+
+		got := NewDecimal128(0, 0)
+		err := json.Unmarshal([]byte(extJSON), &got)
+		assert.Nil(t, err, "json.Unmarshal error: %v", err)
+		assert.Equal(t, want.h, got.h, "expected h: %v got: %v", want.h, got.h)
+		assert.Equal(t, want.l, got.l, "expected l: %v got: %v", want.l, got.l)
+	})
+	t.Run("unmarshal null", func(t *testing.T) {
+		want := NewDecimal128(0, 0)
+		extJSON := `{"$numberDecimal": null}`
+
+		got := NewDecimal128(0, 0)
+		err := json.Unmarshal([]byte(extJSON), &got)
+		assert.Nil(t, err, "json.Unmarshal error: %v", err)
+		assert.Equal(t, want.h, got.h, "expected h: %v got: %v", want.h, got.h)
+		assert.Equal(t, want.l, got.l, "expected l: %v got: %v", want.l, got.l)
+	})
+	t.Run("unmarshal", func(t *testing.T) {
 		cases := append(bigIntTestCases,
 			[]bigIntTestCase{
 				{s: "-0001231.453454000000565600000000E-21", h: 0xafe6000003faa269, l: 0x81cfeceaabdb1800},
 				{s: "12345E+21", h: 0x306a000000000000, l: 12345},
 				{s: "0.10000000000000000000000000000000000000000001", remark: "parse fail"},
 				{s: ".125e1", h: 0x303c000000000000, l: 125},
 				{s: ".125", h: 0x303a000000000000, l: 125},
-				{s: "null", remark: "parse fail"},
```

Comment:
Can we add a test case to ensure unmarshaling the string "null" results in an empty Decimal128, but no error?

Reply:
yup! added on line 173. These test cases put quotes around the string fed in, so I couldn't do it in this block